projects/plugins/jetpack/modules/waf.php

```diff
@@ -11,3 +11,57 @@
  *
  * @package automattic/jetpack
  */
+
+use Automattic\Jetpack\Waf\Waf_Runner;
+use Automattic\Jetpack\Waf\Waf_Standalone_Bootstrap;
+
+/**
+ * Jetpack waf module class.
+ */
+class Jetpack_Waf_Module {
+
+	/**
+	 * Instance of the class.
+	 *
+	 * @var Jetpack_Waf_Module()
+	 */
+	private static $instance = null;
+
+	/**
+	 * Singleton implementation
+	 *
+	 * @return object
+	 */
+	public static function instance() {
+		if ( ! is_a( self::$instance, 'Jetpack_Waf_Module' ) ) {
+			self::$instance = new Jetpack_Waf_Module();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Registers actions
+	 */
+	private function __construct() {
+		add_action( 'jetpack_activate_module_waf', array( $this, 'on_activation' ) );
+		add_action( 'jetpack_deactivate_module_waf', array( $this, 'on_deactivation' ) );
+	}
+
+	/**
+	 * On module activation set up waf mode
+	 */
+	public function on_activation() {
+		update_option( Waf_Runner::MODE_OPTION_NAME, 'normal' );
+		( new Waf_Standalone_Bootstrap() )->generate();
```

Comment:
Don't we just want to call `Waf_Runner->activate()` (and `deactivate()` below) like the CLI does? That does a lot more than this.

Reply:
Added those calls, kept the update_option call as we don't have it on the activate method of Waf_Runner (because we don't want a user that simply activates Jetpack to get the waf running yet)

Reply:
Ah, the option and the bootstrap all happen in `mode()` in the CLI right now. Some slight duplication then but this should be fine.